cmd/mqtt/main.go

```diff
@@ -135,6 +138,24 @@ func main() {
 	}
 	defer nps.Close()
 
+	if cfg.mqttTargetHealthCheck != "" {
+		for {
+			res, err := http.Head(cfg.mqttTargetHealthCheck)
+			if err != nil {
+				logger.Info(fmt.Sprintf("Broker not ready: %s ", err.Error()))
+				time.Sleep(1 * time.Second)
+				continue
+			}
+			if res.StatusCode != 200 {
+				logger.Info(fmt.Sprintf("Broker not ready, status code: %d ", res.StatusCode))
+				time.Sleep(1 * time.Second)
+				continue
+			}
```

Comment:
Looks like you are logging twice

Reply:
First check is if broker is up at all, because adapter will be up before broker (most of the time), and you will get `err` because http request will not be succesfull.
Second check is when broker is up, but is not yet ready to accept connections and not joined cluster. Then you will get status `200` (when broker can accept connections) or `503` when broker is up but not yet ready to accept connections. This is how VerneMQ health check function: 
https://docs.vernemq.com/monitoring/health-check

Reply:
Resolving this one if `err` is null even if status is different than `200`